### PR TITLE
Simplify Parsetree change action and add ppxlib maintainers mention

### PR DESCRIPTION
**Pull request by NathanReb**
I took the liberty to open a PR directly as I have been doing a bit of testing for this action but this is of course open for discussion.

This slighly simplifies the parsetree change action by making use of some of the github actions features and the github-script action to interact with github's API rather than doing it via a separate bash script.

It also includes a few other files to monitor since ppxlib makes uses of other parts of the compiler libraries. In addition to `parsing/parsetree.mli`, it now also adds the label if the following files are modified:
- `parsing/location.mli`
- `parsing/longident.mli`
- `parsing/asttypes.mli`
- `parsing/parse.mli`
- `parsing/pprintast.mli`

Finally, since github notifications don't allow tracking a specific label, the action will ping myself and Patrick whenever a change that can impact ppxlib happens. That will help us stay in touch with the latest state of the compiler and work with the contributors to keep ppxlib's main branch compatible with the development version of OCaml.

### CODE REVIEW

**Comment by NathanReb** [comment]
It is worth noting that the action won't work in its current state as it lacks the correct `permission` settings. Through testing I discovered that github as a somewhat strange way of dealing with authorization with actions and that it seems that the author of the action, or at the very least of the lines adding the `permission` settings has to have write access to the repo.

You can read more about this here: https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository#setting-the-permissions-of-the-github_token-for-your-repository

To put it simply somebody with write access should add the following lines:
```yml
    permissions:
      pull-requests: write
```
following this [line](https://github.com/ocaml/ocaml/pull/13431/files#diff-dd8912f54fc417216736314ed2794042019c05e9cfb7fd909370fb1e7621c8c5R15).

**Comment by NathanReb** [comment]
CC @patricoferris

**Comment by NathanReb** [comment]
CC @panglesd as well who was the original author of the action!

File: .github/workflows/parsetree-change.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    steps:
+      - name: Notify ppxlib maintainers
+        uses: actions/github-script@v7
+        with:
+          script: |
+            github.rest.issues.createComment({
+              issue_number: context.issue.number,
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              body: 'CC @NathanReb @patricoferris'
```
Comment (Octachron) [inline .github/workflows/parsetree-change.yml]: I think it would be a better to create a ppxlib-maintainer team in the `ocaml-ppx` organization and CC that team rather than a fixed set of people.

**Comment by NathanReb** [comment]
Just testing @ocaml-ppx/ppxlib-dev

**Comment by NathanReb** [comment]
I tried creating a team in ocaml-ppx but it seems you cannot mention a team outside of the organisation. Could someone with admin rights in the ocaml organisation create a ppxlib-dev team with @patricoferris and myself?

**Comment by Octachron** [comment]
I have created a `ppxlib-dev` team in the ocaml organization.

**Comment by Octachron** [comment]
I have added the missing permission and tested the workflow at https://github.com/Octachron/ocaml/pull/10 .

**Review by Octachron — APPROVED** [review]
The new workflow action looks simple enough to be low maintenance, thanks !

I wouldn't mind a change entry in the `build system` section of the changelog however.

**Comment by NathanReb** [comment]
Awesome, thanks! I just pushed the changelog entry!
